src/Lecture3.hs

```diff
@@ -105,10 +117,12 @@ newtype Gold = Gold
 
 -- | Addition of gold coins.
 instance Semigroup Gold where
-
+  (<>) :: Gold -> Gold -> Gold
+  Gold { unGold = g1 } <> Gold { unGold = g2 } = Gold { unGold = g1 +g2 }
```

Comment:
You don't need to use records here. It's a nice exercise nevertheless 🤗 
But I think it looks a bit cleaner here with simple pattern matching.
```suggestion
  Gold g1 <> Gold g2 = Gold (g1 + g2)
```